corehunter-base/src/main/java/org/corehunter/objectives/distance/AbstractGenotypeVariantDistanceMetric.java

```diff
@@ -23,17 +23,18 @@
 
 import org.corehunter.data.GenotypeVariantData;
 import uno.informatics.data.SimpleEntity;
+import uno.informatics.data.pojo.DataPojo;
 
 /**
  * @author Guy Davenport
  */
 public abstract class AbstractGenotypeVariantDistanceMetric<GenotypeVariantDataType extends GenotypeVariantData>
-        implements GenotypeVariantDistanceMetric<GenotypeVariantDataType> {
+        extends DataPojo implements GenotypeVariantDistanceMetric<GenotypeVariantDataType> {
 
     private GenotypeVariantDataType data;
 
     public AbstractGenotypeVariantDistanceMetric(GenotypeVariantDataType data) {
-        super();
+        super(data); // TODO review - copies UID, name and data from data
```

Comment:
This will be solved when the objectives have been refactored. Currently the distance metrics extend the data but they will become separate objects that use the data instead (wrapped in a certain aggregation like average or entry-to-nearest-entry).